POEM_072.md

```diff
@@ -0,0 +1,72 @@
+POEM ID: 072  
+Title: Add ability to modify bounds and scaling of implicit outputs and optimizer variables after creation.  
+authors: @robfalck  
+Competing POEMs:  
+Related POEMs:  
+Associated implementation PR: N/A  
+
+
+Status:
+
+- [x] Active
+- [ ] Requesting decision
+- [ ] Accepted
+- [ ] Rejected
+- [ ] Integrated
+
+
+## Motivation
+
+In OpenMDAO, implicit outputs can have bounds attach to them that are enforced by bounds-enforcing linesearches within nonlinear solvers.
+These bounds are defined when `add_output` is called on an implicit component.
+
+In complex models, this means that a user may need to tweak bounds in multiple files.
+The bounds of a variable may be dependent upon the application of the model.
+These bounds may need to be adjusted in a file that is generally outside of the expertise of the user.
+
+Instead, we should take an approach that:
+- allows users to set bounds throughout their model from a single script or notebook.
+- avoids inadvertent git commits that change the model in order to be applicable to the current application.
+
+## What properties of outputs should the user be able to edit outside of the component?
+
+Users should _not_ be able to change variable metadata that affects `compute`, `compute_partials`, or `apply_linear` within a component.
+The internal methods to a component assume a certain set of units, certain shapes, etc.
+So the method we use for adjusting bounds should not be a generic "set_metadata" method.
+
+We should limit the user to changing properties associated with the nonlinear solver: `lower`, `upper`, `ref`, `ref0`, `res_ref`.  I'm currently undecided if we need to give the user the ability to change tags.
+
+When used, OpenMDAO should check, at a minimum, that the given path is a valid output.
+
+## Proposed API for Outputs
+
+```python
+
+prob.model.set_output_solver_options(name='path.to.output', lower=_unspecified, upper=_unspecified, ref=_unspecified, ref0=_unspecified, res_ref=_unspecified)
```

Comment:
That api name would not work with tags, which are not related to solvers

Reply:
Why make this a method on subsystem? Would u expect the user to grab some lower level group or comp? The path.to.var seems to imply that it should be problem level maybe

Reply:
If it's on system the user can use whatever depth they choose.  Doesn't seem to be a good reason to limit them by making it problem level.

Reply:
Yeah I wrestled with the name on this one, for exactly that reason.  We're not allowing the user to override all options, just those pertaining to it's use with solvers.

Reply:
`set_output_solver_options` makes sense to me.

Reply:
I guess it makes sense to me too. I think I initially transposed it in my mind as set_solver_output_options, which is not very good.